### PR TITLE
Test MappingQEulerian with mass and laplace matrix-free operators

### PR DESCRIPTION
**Pull request by davydden**
only add a test for the peace of mind that `MappingQEulerian` works as expected with Laplace operator, namely the mapping is used in the gradient of shape funcitons.

~~WIP as I want to make the test more difficult and also do MPI~~

### CODE REVIEW

File: tests/mappings/mapping_q_eulerian_09.cc [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  mf_laplace.initialize(matrix_free);
+
+  mf_mass.vmult(dst1,src);
+  mf_laplace.vmult(dst3,src);
+
+  // now move manually
+  displacement = 0.;
+  Point<dim> &v = euler_cell->vertex(best_vertex);
+  v(0) += x_displacement;
+  matrix_free->reinit (euler_mapping, dof_handler, constraints, quadrature_formula, data);
```
Comment (davydden) [inline tests/mappings/mapping_q_eulerian_09.cc]: @kronbichler I presume this resets whatever was cached within the `MatrixFree` object with the previous combination of Mapping (u!=0) + Triangulation (unmodified).

Comment (kronbichler) [reply tests/mappings/mapping_q_eulerian_09.cc]: @davydden Yes, this does re-compute the Jacobians etc. that get cached in `MatrixFree`. To minimize the data that is re-computed, one could also set the `MatrixFree::AdditionalData::initialize_indices` to false to keep the previous content, see here:
https://github.com/dealii/dealii/blob/master/tests/matrix_free/update_mapping_only.cc

Comment (davydden) [reply tests/mappings/mapping_q_eulerian_09.cc]: thanks for confirming!

**Comment by davydden** [comment]
I did everything I wanted here

**Pull request merged**